### PR TITLE
ref(*): Upgrade all teh thingz

### PR DESCRIPTION
**Pull request by thomastaylor312**
This PR is massive and I'm sorry. Because we waited so long to upgrade, I ended
having to upgrade everything. This includes the latest versions of the krustlet
crates, tokio 1.0, and all wasmcloud updates. This also includes upgrade
and cleanup of the wasmCloud demos

### CODE REVIEW

File: crates/wasmcloud-provider/Cargo.toml
Comment (thomastaylor312) [inline crates/wasmcloud-provider/Cargo.toml]: This and the next line will be swapped out once the bugfix release of wasmcloud-host is cut

Comment (brooksmtownsend) [reply crates/wasmcloud-provider/Cargo.toml]: @thomastaylor312 did all of your testing work fine with the current `main` branch for wasmcloud-host? If so, we should be ready to bump and release

File: demos/wasmcloud/fileserver/src/lib.rs
Comment (brooksmtownsend) [inline demos/wasmcloud/fileserver/src/lib.rs]: This isn't necessary, but we have a derive macro that allows you to skip the health check definition if you're just going to return healthy, it looks like:
```
#[wasmcloud_actor_core::init]
pub fn init() {
  http::Handlers::register_handle_request(fetch);
}
```
All it does is define a health check that returns healthy for you, so you can avoid defining your own function

File: demos/wasmcloud/greet/src/lib.rs
Comment (brooksmtownsend) [inline demos/wasmcloud/greet/src/lib.rs]: Same comment about init function above

File: demos/wasmcloud/greet/Cargo.toml
Comment (brooksmtownsend) [inline demos/wasmcloud/greet/Cargo.toml]: You shouldn't need the provider core dep for an actor, perhaps this is leftover from an old wascc-codec dependency?

File: demos/wasmcloud/uppercase/src/lib.rs
Comment (brooksmtownsend) [inline demos/wasmcloud/uppercase/src/lib.rs]: same comment as above for the `init` function (again, replacing it with the init won't change the functionality you have, it'll just remove some of the boilerplate you have to put in for the health check)

File: demos/wasmcloud/uppercase/Cargo.toml
Comment (brooksmtownsend) [inline demos/wasmcloud/uppercase/Cargo.toml]: `wasmcloud-provider-core` dependency here can be removed

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
All the things look good! My only comments concern leaving the `wasmcloud-provider-core` dependency in the Cargo.toml and the new derive macro we have for `wasmcloud-actor-core`'s init function, both of those will trim down the demos you all have a bit so people can only focus on the **BUSINESS LOGIC**.

File: crates/wasmcloud-provider/Cargo.toml
Comment (thomastaylor312) [reply crates/wasmcloud-provider/Cargo.toml]: Yep, this is good to go

Comment (thomastaylor312) [reply crates/wasmcloud-provider/Cargo.toml]: Yep, this is good to go

File: demos/wasmcloud/fileserver/src/lib.rs
Comment (thomastaylor312) [reply demos/wasmcloud/fileserver/src/lib.rs]: I'm all for shortening (both code and the cooking fat). I'll fix it right now

File: demos/wasmcloud/greet/Cargo.toml
Comment (thomastaylor312) [reply demos/wasmcloud/greet/Cargo.toml]: More than likely that is correct

File: crates/wasmcloud-provider/Cargo.toml
Comment (bacongobbler) [inline crates/wasmcloud-provider/Cargo.toml]: note to not merge until the new wasmcloud-host version is out.

File: crates/wasmcloud-provider/src/lib.rs
Comment (bacongobbler) [inline crates/wasmcloud-provider/src/lib.rs:391]: nit: should this be shortened to `provider_id`?

File: crates/wasmcloud-provider/src/lib.rs
Comment (bacongobbler) [inline crates/wasmcloud-provider/src/lib.rs:524]: Are there any security implications we need to consider here? Not familiar with the new wash claims stuff. Just want to make sure we're doing the right thing by hardcoding these tokens into the server. I'm wondering if these should be generated with a build.rs file for example?

File: demos/wasmcloud/greet/greet-wasmcloud.yaml
Comment (bacongobbler) [inline demos/wasmcloud/greet/greet-wasmcloud.yaml:9]: 👍 for proper semver

File: src/krustlet-wasmcloud.rs
Comment (bacongobbler) [inline src/krustlet-wasmcloud.rs:9]: why the change to actix?

**Review by bacongobbler — COMMENTED** [review]
few comments here and there but looks great!

File: crates/wasmcloud-provider/src/lib.rs
Comment (thomastaylor312) [reply crates/wasmcloud-provider/src/lib.rs:391]: I thought that would lead to confusion with the fact that this is _also_ a provider, but of the Krustlet variety

File: crates/wasmcloud-provider/src/lib.rs
Comment (thomastaylor312) [reply crates/wasmcloud-provider/src/lib.rs:524]: According to @brooksmtownsend, this is appropriate here. Doing it with `build.rs` is an interesting idea, but I would worry that regenerating the keys each time it is built isn't the right thing. Also, they'd still be a const in the final binary and you could likely get to them with `strings`

File: src/krustlet-wasmcloud.rs
Comment (thomastaylor312) [reply src/krustlet-wasmcloud.rs:9]: It was needed from the wasmCloud side of things. Actix is actually just wrapping around the normal tokio stuff, it starts a normal tokio runtime and starts the Actix `System` wasmCloud needs

File: crates/wasmcloud-provider/src/lib.rs
Comment (bacongobbler) [reply crates/wasmcloud-provider/src/lib.rs:391]: ooh yeah good call. I guess a little diasmbiguation is good here.

**Review by bacongobbler — APPROVED** [review]
I will provisionally LGTM here. It would be great if @brooksmtownsend could clarify a bit on https://github.com/wasmCloud/krustlet-wasmcloud-provider/pull/1#discussion_r610075217 just to be absolutely sure we're doing the right thing here, but I'll leave it as a non-blocking comment. Otherwise this looks good to go (pending wamscloud-host update).

File: crates/wasmcloud-provider/src/lib.rs
Comment (bacongobbler) [reply crates/wasmcloud-provider/src/lib.rs:524]: My thinking with build.rs here is: if someone forks krustlet-wasmcloud-provider and makes changes to the internal logging crate, would they need to regenerate the public key with their unique vendor name or something?

Comment (brooksmtownsend) [reply crates/wasmcloud-provider/src/lib.rs:524]: So as far as security implications, there are none for keeping the pubkey / JWT in the code, it's all public information and you don't have to worry about it. The `wash claims` stuff works the same way that `wascap` did so all concepts follow. The thing here about having the providers loaded as static plugins is that these claims can be re-used, the only time you would have to regenerate and be careful about generating them each time would be if you were creating your own provider archive.

Comment (brooksmtownsend) [reply crates/wasmcloud-provider/src/lib.rs:524]: I suppose TL;DR here is that loading providers as static plugins still require claims for linking, but these embedded tokens could be re-used by other people just fine here. I'd be more concerned with potentially including the private keys so that people could generate the same JWT just in case those weren't changed in the future.

File: crates/wasmcloud-provider/Cargo.toml
Comment (brooksmtownsend) [reply crates/wasmcloud-provider/Cargo.toml]: new wasmcloud-host version is out as of now 😄 try `0.16.1`

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**
